File: fixtures/scale/python/large/cipherlab/checks/check_conveyor.py
from cipherlab.ciphers.conveyor import conveyor_encrypt, conveyor_decrypt
from cipherlab.specs.conveyor_spec import conveyor_spec
from cipherlab.corpus.corpus_rotate import corpus_rotate


def check_conveyor():
    spec = conveyor_spec()
    for text in corpus_rotate():
        sealed = conveyor_encrypt(text, spec["key"])
        opened = conveyor_decrypt(sealed, spec["key"])
        if opened != text:
            return False
    return spec["category"] == "rotate"

File: fixtures/scale/ruby/large/lib/check_conveyor.rb
require_relative 'conveyor_encrypt'
require_relative 'conveyor_decrypt'
require_relative 'corpus_rotate'

def check_conveyor
  corpus_rotate.each do |text|
    sealed = conveyor_encrypt(text, 19)
    opened = conveyor_decrypt(sealed, 19)
    return false if opened != text
  end
  true
end
